### ISSUE TITLE
Crash: Boxplots

### ISSUE THREAD

**Issue by GeoDaTester** [GeoDaCenter/geoda#1055]
From: wtassinari@gmail.com
 Details:
os: 3-4-4
vs: 1-10-0-8
/usr/local/geoda/web_plugins/logger.txt
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Check auto update:
AutoUpdate::CheckUpdate()
AutoUpdate::GetCheckList()
AutoUpdate::ReadUrlContent()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/media/tassinari/Seagate Slim Drive/FIOCRUZ/Lidiane/RS.shp
Open Layer:RS
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/media/tassinari/Seagate Slim Drive/FIOCRUZ/Lidiane/RS.shp
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In GdaFrame::OpenQuantile(4)
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_4
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_4
Open BoxPlotCanvas.
Close BoxPlotCanvas.
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_1
maformco_2
Open ScatterNewPlotFrame.
MapCanvas::~MapCanvas()
MapCanvas::~MapCanvas()
Click GdaFrame::OnCloseProjectEvt
Project::GetProjectFullPath()
In TableFrame::OnClose and actually closing.
Project::CleanupPairsHLState()
Entering Project::~Project
Exiting Project::~Project
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/media/tassinari/Seagate Slim Drive/FIOCRUZ/Lidiane/areaEstudo.shp
Open Layer:areaEstudo
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/media/tassinari/Seagate Slim Drive/FIOCRUZ/Lidiane/areaEstudo.shp
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In GdaFrame::OpenQuantile(5)
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_1
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Click GdaFrame::OnLegendBackgroundColor
Click GdaFrame::OnLegendBackgroundColor
Click GdaFrame::OnLegendBackgroundColor
Click GdaFrame::OnLegendBackgroundColor
Click GdaFrame::OnCanvasBackgroundColor
MapCanvas::~MapCanvas()
In GdaFrame::OnOpenHinge15()
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_2
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
MapCanvas::~MapCanvas()
In TableFrame::OnActivate
In TableFrame::OnLabelLeftClickEvent()
In TableFrame::OnLabelLeftClickEvent()
In TableFrame::OnLabelLeftClickEvent()
In TableFrame::OnLabelLeftClickEvent()
Open LisaScatterPlotFrame (OnOpenMoranEB).
Click GdaFrame::OnToolsWeightsManager
Entering WeightsManFrame::WeightsManFrame
In WeightsManFrame::SetDetailsForItem
Exiting WeightsManFrame::WeightsManFrame
In WeightsManFrame::OnActivate
In WeightsManFrame::OnCreateBtn
Click GdaFrame::OnToolsWeightsCreate
Open CreatingWeightDlg
Click CreatingWeightDlg::OnCThresholdTextEdit:
In WeightsManFrame::OnActivate
Click CreatingWeightDlg::OnIdVariableSelected
Click CreatingWeightDlg::OnCreateNewIdClick
Try to insert unique id (integer, POLY_ID) column in table.
Click CreatingWeightDlg::OnCRadioRookSelected
Click CreatingWeightDlg::OnCreateClick
CreateWeights()/media/tassinari/Seagate Slim Drive/FIOCRUZ/Lidiane/areaEstudo.gal
In WeightsManFrame::update(WeightsManState* o)
In WeightsManFrame::OnWListItemSelect
In WeightsManFrame::SetDetailsForItem
In WeightsManFrame::update(WeightsManState* o)
In WeightsManFrame::OnActivate
WeightsManFrame::OnHistogramBtn()
Open ConnectivityHistFrame
In WeightsManFrame::OnActivate
ConnectivityHistFrame::OnActivate
In WeightsManFrame::OnActivate
WeightsManFrame::OnConnectMapBtn()
Open ConnectivityMapFrame
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In WeightsManFrame::OnActivate
In ConnectivityMapFrame::OnActivate
MapCanvas::~MapCanvas()
In WeightsManFrame::OnActivate
Open LisaScatterPlotFrame (OnOpenMSPL).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_2
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
Open LisaMapFrame (OnOpenUniLisa).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_2
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Click GdaFrame::OnCopyLegendToClipboard
In TableFrame::OnActivate
In TableFrame::OnLabelLeftClickEvent()
Open LisaMapFrame (OnOpenLisaEB).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_1
maformco_2
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Open LisaScatterPlotFrame (OnOpenMoranEB).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_1
maformco_2
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
VariableSettingsDlg::GetWeightsId() weight: areaEstudo
In TableFrame::OnActivate
Open FieldNewCalcSheetDlg.
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnPageChange()
Open DataViewerAddColDlg.
DataViewerAddColDlg::OnEditName()
DataViewerAddColDlg::OnEditDecimals()
DataViewerAddColDlg::OnEditLength()
DataViewerAddColDlg::OnEditDecimals()
DataViewerAddColDlg::OnEditName()
DataViewerAddColDlg::OnEditName()
DataViewerAddColDlg::OnEditName()
DataViewerAddColDlg::OnEditName()
DataViewerAddColDlg::OnEditName()
Entering DataViewerAddColDlg::OnOkClick
Inserting new column BAYES into Table
Click CreatingWeightDlg::OnCThresholdTextEdit:
0.0
Click GdaFrame::OnToolsWeightsManager
Entering WeightsManFrame::WeightsManFrame
In WeightsManFrame::SetDetailsForItem
Exiting WeightsManFrame::WeightsManFrame
In WeightsManFrame::OnActivate
In FieldNewCalcSheetDlg::OnApplyClick()
Click CreatingWeightDlg::OnCThresholdTextEdit:
0.0
In FieldNewCalcSheetDlg::OnClose()
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
Open FieldNewCalcSheetDlg.
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnPageChange()
In FieldNewCalcSheetDlg::OnClose()
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformco_3
Open BoxPlotCanvas.
Close BoxPlotCanvas.
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
maformcong
Open BoxPlotCanvas.

**Comment by jkoschinsky** [comment]
Julia will try to replicate.

Steps:

1. quantile map (4)
2. box plot
3. quantile (5)
4. change background color
5. boxmap - 1.5 hinge
6. LISA
7. weights
8. add unique ID
9. weights histogram
10. connectivity map
11. LISA
12. copy legend to clipboard
13. open LISA EB
14. add column to table: Bayes
15. open boxplot

**Comment by VeraB620** [comment]
OS X El Capitan 10.11.6
GeoDa 1.12.1.147

Following jkoschinsky steps, I had no crash.
I used 4 different data sets (2 from GeoDa, 2 from Chicago Portal), keeping all files open until end and then again closing each file following the crash report, using new variable for each map/plot and then using same variable. I had no lag or crash. 

The only thing that seemed odd was according to the GeoDa documentation I should have received an error when trying to change string to date, but I did not get that error, it worked OK (not sure if this is a new feature)
<img width="895" alt="screen shot 2018-08-13 at 3 46 11 pm" src="https://user-images.githubusercontent.com/4943475/44057354-8e1927ce-9f10-11e8-82d0-36a235ed4310.png">
